### PR TITLE
refactor/font-ratio-rhythm-prim-s

### PR DESCRIPTION
**Pull request by dan-entous**
this pr effects the _font-size_, _line-height_, _margin-top_, and _margin-bottom_ in the following components. nothing should change in comparison with production:
* auto-complete
* business-card
* footer-navigation
* table-compare
* link
* content-manageable
* notification
* preferences-bar

### CODE REVIEW

File: src/scss/tools/mixins/_typography.scss [hunk 1]
```diff
@@ -21,3 +21,13 @@
     @include font-line-ratio($sizeName);
     @include font-rhythm($sizeName);
 }
+
+// step-by-step eliminate this insanity
+// replaces @extend .copy--small;
+// replaces @include font-ratio-rhythm(prim-s);
+@mixin font-ratio-rhythm-prim-s {
```
Comment (matijs) [inline src/scss/tools/mixins/_typography.scss]: It's not actually used :) so this can 👋

Comment (dan-entous) [reply src/scss/tools/mixins/_typography.scss]: 1. wanted to take a step-by-step approach and not initially mess with the typography robin and plas setup, but we can discuss at the office: i hate having to use the negative margins this way to achieve "vertical rhythm".
1. wanted to also discuss the best approach, pull in the mixin or paste the result of the mixin. found a few conflicting styles, so i pasted the result in and eliminated the excess styling.

File: src/js/components/molecules/auto-complete/auto-complete.scss [hunk 2]
```diff
@@ -30,7 +30,10 @@
     }
 
     &__list-item {
-        @extend .copy--small;
+        font-size: 14px;
+        line-height: 1.5;
+        margin-top: -2px;
```
Comment (matijs) [inline src/js/components/molecules/auto-complete/auto-complete.scss]: I'd remove this.

File: src/js/components/molecules/auto-complete/auto-complete.scss [hunk 3]
```diff
@@ -30,7 +30,10 @@
...
+        margin-bottom: 23px;
```
Comment (matijs) [inline src/js/components/molecules/auto-complete/auto-complete.scss]: And make this `margin-bottom: 21px;`

Comment (dan-entous) [reply src/js/components/molecules/auto-complete/auto-complete.scss]: i started to look into it and found yet another rabbit hole. i couldn't find auto-complete in the site, so i looked at business-card. some other style sets margin-top to 0 negating this setting, which overrides a -8px, which overrides yet another setting from another class ...

i think we should initially replace these extends and then go over the components and correct them.

**Pull request merged**